src/intersection.class.ts

```diff
@@ -1,68 +1,85 @@
 //@ts-nocheck
-import { Point } from './point.class';
+import { Point } from "./point.class";
+import { fabric } from '../HEADER';
 
-(function(global) {
-  /* Adaptation of work of Kevin Lindsey (kevin@kevlindev.com) */
-  var fabric = global.fabric || (global.fabric = { });
-  /**
-   * Intersection class
-   * @class fabric.Intersection
-   * @memberOf fabric
-   * @constructor
-   */
-  function Intersection(status) {
+/* Adaptation of work of Kevin Lindsey (kevin@kevlindev.com) */
+
+type IntersectionType = 'Intersection' | 'Coincident' | 'Parallel';
+
+/**
+ * **Assuming `T`, `A`, `B` are points on the same line**,
+ * check if `T` is contained in `[A, B]` by comparing the direction of the vectors from `T` to `A` and `B`
+ * @param T 
+ * @param A 
+ * @param B 
+ * @returns true if `T` is contained
+ */
+const isContainedInInterval = (T: Point, A: Point, B: Point) => {
```

Comment:
I am afraid future misuse of the method because it is meant to be used in a very specific context as mentioned in the comment

Any suggestions?